### ISSUE TITLE
[Feature Request] Add support for JSDoc "import()" feature

### ISSUE THREAD

**Issue by aeolingamenfel** [bradzacher/eslint-plugin-typescript#154]
TypeScript (and VSCode) have support for a JSDoc syntax that allows you to `import()` directly within a JSDoc type. Example:

```Javascript
/**
 * @param {import("vuex").ActionContext} val1
 */
function myFun(val1) {
    // ...
}
```

(VueX example just because I can get my intellisense in VSCode to work with that as a working example)

This allows a user to reference an object if it isn't imported (like `ActionContext` in this example). 

Would love to see some sort of support for this syntax in this plugin. Specifically, this syntax currently fails by violating the ESLint `valid-jsdoc` rule as a syntax error. Ideally, support for this would flag this as valid syntax.

---

You can find some references to the `import()` syntax (couldn't find formal documentation easily) on Microsoft/TypeScript#14844.

**Comment by bradzacher** [comment]
could be a candidate for this plugin as it is TS specific syntax.

Just so I can understand the use case for this.
How come you're using the import syntax in your comment instead of using proper typescript types?

Are you gradually adding types to a `js` file?

**Comment by bradzacher** [comment]
As per https://eslint.org/blog/2018/11/eslint-v5.9.0-released
ESLint has EOL JSDoc support.

> The `require-jsdoc` and `valid-jsdoc` rules will be deprecated. These two rules will remain in ESLint but we will no longer add new features or fix bugs for them. These rules may be removed in a future major release of ESLint.

As such, this is no longer a good candidate for us to build into this plugin for two main reasons:
1) We'd have to fork and maintain the entire base logic as they will remove it in an upcoming release.
2) We'd have to fork and maintain Doctrine (the jsdoc parser).

Considering most of our users use this plugin for *typescript*, which interops with jsdoc to provide types without explicitly documenting them in a comment.

Might be a feature candidate instead for https://github.com/gajus/eslint-plugin-jsdoc.
